engine/src/main/java/org/terasology/logic/players/LocalPlayerSystem.java

```diff
@@ -222,28 +227,21 @@ private void standPlayer(EntityRef entity) {
      */
     private void stopAutoMove() {
         isAutoMove = false;
+        List<Input> inputs = bindsConfig.getBinds(new SimpleUri("engine:forwards"));
+        Input forwardKey = inputs.get(0);
+        inputSystem.cancelSimulatedKeyStroke(forwardKey);
     }
 
     /**
      * Start a periodic action that moves the entity in the camera direction at every period of 0.2 seconds.
      */
-    private void startAutoMove(EntityRef entity) {
+    private void startAutoMove() {
         isAutoMove = true;
-        delayManager.addPeriodicAction(entity, AUTO_MOVE_ID, 0, 200);
-    }
-
-    @ReceiveEvent
-    public void onAutoMove(PeriodicActionTriggeredEvent event, EntityRef entity) {
-        if (event.getActionId().equals(AUTO_MOVE_ID)) {
-            if (isAutoMove) {
-                ClientComponent clientComponent = entity.getComponent(ClientComponent.class);
-                Vector3f viewDir = entity.getComponent(LocationComponent.class).getWorldDirection();
-                clientComponent.character.send(new CharacterImpulseEvent(viewDir.mul(8f)));
-            } else {
-                // If isAutoMove turns false, cancel the action.
-                delayManager.cancelPeriodicAction(entity, AUTO_MOVE_ID);
-            }
-        }
+        bindsConfig = config.getInput().getBinds();
+        List<Input> inputs = bindsConfig.getBinds(new SimpleUri("engine:forwards"));
+        Input forwardKey = inputs.get(0);
+        inputSystem.simulateSingleKeyStroke(forwardKey);
+        inputSystem.simulateRepeatedKeyStroke(forwardKey);
```

Comment:
is it always necessary to trigger a single key stroke before a repeated key stroke? if so, this should be documented in the Javadoc, otherwise triggering a repeated key stroke should be sufficient(?)